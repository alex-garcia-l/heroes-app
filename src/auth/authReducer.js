import { authType } from '../types/auth';

export const authReducer = (state = {}, action) => {
  
  switch (action.type) {
    case authType.login:
      return {
        ...action.payload,
        logged: true
      }
    case authType.logout:
      return {
        logged: false
      }
  
    default:
      return state;
  }

}