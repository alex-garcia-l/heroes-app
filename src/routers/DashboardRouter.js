import React from 'react';
import { Route, Routes } from 'react-router-dom';

import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heros/HeroScreen';
import { HomeScreen } from '../components/HomeScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';

import { Navbar } from '../components/ui/Navbar';

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-4">
        <Routes>
          <Route exact index path="/" element={<HomeScreen />} />
          <Route exact path="marvel" element={<MarvelScreen />} />
          <Route exact path="search" element={<SearchScreen />} />
          <Route exact path="hero/:id" element={<HeroScreen />} />
          <Route exact path="dc" element={<DcScreen />} />
        </Routes>
      </div>
    </>
  );
};
