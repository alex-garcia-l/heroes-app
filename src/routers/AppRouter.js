import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRouter } from './DashboardRouter';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>

        <Route
          exact
          path="login"
          element={
            <PublicRoute>
              <LoginScreen />
            </PublicRoute>
          }
        />

        <Route path="/*" 
          element={
            <PrivateRoute>
              <DashboardRouter />
            </PrivateRoute>
          }
        />
      </Routes>
    </BrowserRouter>
  );
};
