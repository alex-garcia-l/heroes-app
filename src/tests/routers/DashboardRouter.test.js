import { mount } from 'enzyme';
import { extract } from 'query-string';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { DashboardRouter } from '../../routers/DashboardRouter';

describe('Prueba para el componente <DashboardRouter />', () => {

  const contextValue = {
    user: {
      name: 'Name',
      logged: true
    }
  }

  test('debe mostrarse correctamente la ruta "/home"', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/home']} >
          <DashboardRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(extract).toMatchSnapshot();
    expect(wrapper.find('span.me-4').text().trim()).toBe(contextValue.user.name);
    expect(wrapper.find('h2').first().text().trim()).toBe('Search')
  });

  test('debe mostrarse correctamente la ruta "/marvel"', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/marvel']} >
          <DashboardRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(extract).toMatchSnapshot();
    expect(wrapper.find('h2').first().text().trim()).toBe('Marvel Heros')
  });

  test('debe mostrarse correctamente la ruta "/dc"', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/dc']} >
          <DashboardRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(extract).toMatchSnapshot();
    expect(wrapper.find('h2').first().text().trim()).toBe('DC Heros')
  });

});
