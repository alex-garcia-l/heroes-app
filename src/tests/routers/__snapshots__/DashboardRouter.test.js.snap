// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para el componente <DashboardRouter /> debe mostrarse correctamente la ruta "/dc" 1`] = `[Function]`;

exports[`Prueba para el componente <DashboardRouter /> debe mostrarse correctamente la ruta "/home" 1`] = `[Function]`;

exports[`Prueba para el componente <DashboardRouter /> debe mostrarse correctamente la ruta "/marvel" 1`] = `[Function]`;
